meson_options.txt

```diff
@@ -38,3 +38,4 @@ option('efi-libdir', type : 'string', description : 'path to the EFI lib directo
 option('efi-ldsdir', type : 'string', description : 'path to the EFI lds directory')
 option('efi-includedir', type : 'string', value : '/usr/include/efi', description : 'path to the EFI header directory')
 option('efi_os_dir', type: 'string', description : 'the name of OS directory in ESP')
+option('link_language', type : 'string', value : 'c', description : 'the linker to use for targets [c, cpp]')
```

Comment:
rather than a new option to force this, can you detect the compiler or environment in meson.build?

Reply:
> can you detect the compiler or environment in meson.build

I tried compiler, but just got `clang`. I think env vars are frowned upon for building, e.g. https://github.com/mesonbuild/meson/issues/9

Reply:
Hmm OK.  Well it would be better not to have an option to activate a hack.  How about if you just use a cross compilation script and set the linker?  See https://github.com/mesonbuild/meson/pull/6207

Reply:
Actually you might not even need a cross file looking at that, it seems like if you set `LD` in environment it should respect it in meson 0.53.0+

Reply:
Hmm, I tried `export LD=$CXX` but it seemed to not use the linker. I'll try a cross.ini and see if that can work.

Reply:
Hmm, no dice:

    [binaries]
    c_ld = 'clang++'

gave me

    stderr: clang-12: error: invalid linker name in argument '-fuse-ld=clang++'

Reply:
Isn't that meaning it's a clang bug not a meson bug though?  Or is it maybe not in PATH and you need to give a full path?

Reply:
From the meson maintainer `Yep, wont work. It just passes -fuse-ld=yourthing to the linker.`